src/output.cc

```diff
@@ -43,171 +42,199 @@
 #endif
 #include "output.h"
 
-static struct output_module *modules[] = {
+#define TAG "output"
+
+static OutputModule* output_module = NULL;
+
+/**
+  @brief  Describes an available output module by name, description. Provides an
+  interface for construction of the output and a means to query the output for
+  associated command line options.
+*/
+struct OutputEntry {
+  std::string shortname;
+  std::string description;
+  std::function<OutputModule*(Output::PlaybackCallback,
+                              Output::MetadataCallback)> create;
+  OutputModule::Options& options;
+};
+
+static std::vector<OutputEntry>& GetAvailableModules() {
```

Comment:
You probably want to return a `const std::vector<OutputEntry>&` here, as you don't want the caller have it modify.